sklearn/utils/tests/test_validation.py

```diff
@@ -349,6 +349,31 @@ def test_check_array():
             check_array(X, dtype="numeric")
 
 
+@pytest.mark.parametrize("dtype", ["Int8", "Int16", "Int32",
+                                   "UInt8", "UInt16", "UInt32"])
+def test_check_array_pandas_na_support(dtype):
+    # Test pandas IntegerArray with pd.NA
+    pd = pytest.importorskip('pandas', minversion="1.0")
+
+    X_np = np.array([[1, 2, 3, np.nan, np.nan],
+                     [np.nan, np.nan, 8, 4, 6],
+                     [1, 2, 3, 4, 5]]).T
+
+    # Creates dataframe with IntegerArrays with pd.NA
+    X = pd.DataFrame(X_np, dtype=dtype, columns=['a', 'b', 'c'])
+    # column c has no nans
+    X['c'] = X['c'].astype('float')
+    X_checked = check_array(X, force_all_finite='allow-nan')
```

Comment:
Maybe you could extend this test to have variant where you pass `dtype=[np.float32, np.float64]` and you check that `X_checked.dtype == np.float32`.